examples/energy-management-app/energy-management-common/src/EnergyEvseMain.cpp

```diff
@@ -0,0 +1,411 @@
+    {
+        ChipLogError(AppServer, "Failed to allocate memory for EvseManufacturer");
+        return CHIP_ERROR_NO_MEMORY;
+    }
+
+    /* Call Manufacturer specific init */
+    err = gEvseManufacturer->Init();
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(AppServer, "Init failed on gEvseManufacturer");
+        gEvseManufacturer.reset();
+        return err;
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR EVSEManufacturerShutdown()
+{
+    if (gEvseManufacturer)
+    {
+        /* Shutdown the EVSEManufacturer */
+        gEvseManufacturer->Shutdown();
+        gEvseManufacturer.reset();
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+void EvseApplicationInit()
+{
+    if (DeviceEnergyManagementInit() != CHIP_NO_ERROR)
+    {
+        return;
+    }
+
+    if (EnergyEvseInit() != CHIP_NO_ERROR)
+    {
+        DeviceEnergyManagementShutdown();
+        return;
+    }
+
+    if (EnergyMeterInit() != CHIP_NO_ERROR)
+    {
+        DeviceEnergyManagementShutdown();
+        EnergyEvseShutdown();
+        return;
+    }
+
+    if (EVSEManufacturerInit() != CHIP_NO_ERROR)
+    {
+        DeviceEnergyManagementShutdown();
+        EnergyEvseShutdown();
+        EnergyMeterShutdown();
+        return;
+    }
+}
+
+void EvseApplicationShutdown()
+{
+    ChipLogDetail(AppServer, "Energy Management App: ApplicationShutdown()");
+
+    /* Shutdown in reverse order that they were created */
+    EVSEManufacturerShutdown();       /* Free the EVSEManufacturer */
+    EnergyMeterShutdown();            /* Free the Energy Meter */
+    EnergyEvseShutdown();             /* Free the EnergyEvse */
+    DeviceEnergyManagementShutdown(); /* Free the DEM */
+
+    Clusters::DeviceEnergyManagementMode::Shutdown();
+    Clusters::EnergyEvseMode::Shutdown();
+}
+
+void emberAfElectricalEnergyMeasurementClusterInitCallback(chip::EndpointId endpointId)
+{
+    VerifyOrDie(endpointId == 1); // this cluster is only enabled for endpoint 1.
+    VerifyOrDie(!gEEMAttrAccess);
+
+    gEEMAttrAccess = std::make_unique<ElectricalEnergyMeasurementAttrAccess>(
+        BitMask<ElectricalEnergyMeasurement::Feature, uint32_t>(
+            ElectricalEnergyMeasurement::Feature::kImportedEnergy, ElectricalEnergyMeasurement::Feature::kExportedEnergy,
+            ElectricalEnergyMeasurement::Feature::kCumulativeEnergy, ElectricalEnergyMeasurement::Feature::kPeriodicEnergy),
+        BitMask<ElectricalEnergyMeasurement::OptionalAttributes, uint32_t>(
+            ElectricalEnergyMeasurement::OptionalAttributes::kOptionalAttributeCumulativeEnergyReset));
+
+    // Create an accuracy entry which is between +/-0.5 and +/- 5% across the range of all possible energy readings
+    ElectricalEnergyMeasurement::Structs::MeasurementAccuracyRangeStruct::Type energyAccuracyRanges[] = {
+        { .rangeMin   = 0,
+          .rangeMax   = 1'000'000'000'000'000, // 1 million Mwh
+          .percentMax = MakeOptional(static_cast<chip::Percent100ths>(500)),
```

Comment:
Might or might not be more readable as: `MakeOptional<Percent100ths>(500)` (and similar for MakeNullable below...).

Reply:
Ah, well.  I was hoping the compiler would be smarter about it.  Understood.